execution/src/deposited.rs

```diff
@@ -95,33 +76,48 @@ pub(super) fn execute_deposited_transaction<
             &mut gas_meter,
             &mut traversal_context,
         )
-        .map_err(Into::into);
-
-    let mint_params = outcome.and_then(|values| {
-        let evm_outcome = extract_evm_result(values);
-        if !evm_outcome.is_success {
-            return Err(UserError::DepositFailure(evm_outcome.output));
-        }
-        let mint_params = get_mint_params(&evm_outcome);
-        Ok((mint_params, evm_outcome.logs))
-    });
-
-    let (logs, vm_outcome) = match mint_params {
-        Ok((Some(mint_params), logs)) => {
-            eth_token::mint_eth(
-                &mint_params.destination,
-                mint_params.amount,
+        .map_err(Error::from)
+        .and_then(|values| {
+            let evm_outcome = extract_evm_result(values);
+            if !evm_outcome.is_success {
+                return Err(UserError::DepositFailure(evm_outcome.output).into());
+            }
+
+            // If there is a non-zero mint amount then we start by
+            // giving those tokens to the EVM native address.
+            // The tokens will then be distributed to the correct
+            // accounts according to the transferred that happened
```

Comment:
nit: transferred -> transfers

Reply:
Fixed in https://github.com/MovedNetwork/op-move/pull/319/commits/3a899d67ceb68cc39e56b799778256fe6d05d31c